src/main/java/com/bootcamp/demo/controller/PaymentController.java

```diff
@@ -0,0 +1,36 @@
+package com.bootcamp.demo.controller;
+
+import com.bootcamp.demo.exception.PaymentFailException;
+import com.bootcamp.demo.model.Card;
+import com.bootcamp.demo.model.Transaction;
+import com.bootcamp.demo.service.PaymentService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+import java.time.LocalDateTime;
+
+@RestController
+@RequestMapping("/payment")
+public class PaymentController {
+
+    private final PaymentService paymentService;
+
+    @Autowired
+    public PaymentController(PaymentService paymentService) {
+        this.paymentService = paymentService;
+    }
+
+    @GetMapping("/test")
+    public Transaction getTestTransaction(){
+        Transaction transaction = new Transaction("123435", "657353845", 234.43,  LocalDateTime.now());
+        return transaction;
+    }
+
+    @PostMapping("/pay")
+    public Transaction pay(Card card, Double amount) throws PaymentFailException {
```

Comment:
In order to use a `@PostMapping` with a request body the method's parameter must be annotated with `@RequestBody`. If you want multiple parameters in that request body you must wrap them in another object (e.g. TransactionRequest) that can be successfully deserialized from the JSON Request Body.
Please see more details about this top [here](https://docs.spring.io/spring-framework/docs/current/reference/html/web.html#mvc-ann-requestbody), [here ](https://www.baeldung.com/spring-request-response-body) and [here](https://stackoverflow.com/questions/11291933/requestbody-and-responsebody-annotations-in-spring).

Reply:
Updated base branch to feature/payment-by-card instead of master. Those are not my modifications